### ISSUE TITLE
False negative for nested media queries in selector-max-compound-selectors

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2136]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Expected a different behavior.

Ref: https://github.com/stylelint/stylelint/pull/2102#issue-191053388

> Which rule, if any, is this issue related to?

`selector-max-compound-selectors`

> What CSS is needed to reproduce this issue?

e.g.

```scss
.foo .bar > .foobar { 
  & .test { 
    @media screen { 
      color: red; 
    }
  } 
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-max-compound-selectors": "3"
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

No matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS

> What did you expect to happen?

Throw error.

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing
